qiskit/quantum_info/states/stabilizerstate.py

```diff
@@ -196,15 +196,21 @@ def evolve(self, other, qargs=None):
         return ret
 
     def expectation_value(self, oper, qargs=None):
-        """Compute the expectation value of an operator.
+        """Compute the expectation value of a Pauli operator.
 
         Args:
-            oper (BaseOperator): an operator to evaluate expval.
+            oper (Pauli): a Pauli operator to evaluate expval.
             qargs (None or list): subsystems to apply the operator on.
 
         Returns:
-            complex: the expectation value (only 0 or 1 or -1).
+            complex: the expectation value (only 0 or 1 or -1 or i or -i).
+
+        Raises:
+            QiskitError: if oper is not a Pauli operator.
         """
+        if not isinstance(oper, Pauli):
+            raise QiskitError("Operator for Expetation value is not a Pauli operator.")
```

Comment:
In principle this is a breaking change, but I think in practice this function already required the input to be a `Pauli`, via the `oper.to_label()` call further down. Is that right?

Reply:
Indeed, in practice the current code worked only for (unsigned) Paulis, so this just asserts that the given input in indeed a Pauli element.